yellowbox/extras/redis.py

```diff
@@ -1,22 +1,29 @@
-from typing import TypeVar, Callable
+from typing import IO, TypeVar, Callable
 
 from docker import DockerClient
 from redis import ConnectionError as RedisConnectionError, Redis
 
-from yellowbox.containers import get_ports, create_and_pull
+from yellowbox.containers import get_ports, create_and_pull, upload_file
 from yellowbox.subclasses import SingleContainerService, RunnableWithContext
 from yellowbox.utils import retry
 
 REDIS_DEFAULT_PORT = 6379
+DEFAULT_RDB_PATH = "/data/dump.rdb"
+
 _T = TypeVar("_T")
 
 
 class RedisService(SingleContainerService, RunnableWithContext):
-    def __init__(self, docker_client: DockerClient, image='redis:latest', **kwargs):
-        super().__init__(
-            create_and_pull(docker_client, image, publish_all_ports=True, detach=True),
-            **kwargs
-        )
+    def __init__(self, docker_client: DockerClient, image='redis:latest',
+                 redis_file: bytes = None, **kwargs):
+        container = create_and_pull(docker_client, image, publish_all_ports=True, detach=True)
+        self.started = False
```

Comment:
maybe this should be standard in all containerservices?

Reply:
dunno 🤷